proofs/eo/cpc/rules/Quantifiers.eo

```diff
@@ -127,7 +127,8 @@
   (($mk_quant_unused_vars_rec @list.nil F)    @list.nil)
   (($mk_quant_unused_vars_rec (@list x xs) F) (eo::define ((r ($mk_quant_unused_vars_rec xs F)))
                                               (eo::ite ($contains_subterm F x)
-                                                (eo::ite ($contains_subterm r x) r (eo::cons @list x r))
+                                                ; remove the duplicate of x in r if it exists
+                                                (eo::cons @list x ($nary_remove @list @list.nil x r))
```

Comment:
This is ok. It's probably simpler and more efficient to add the list `r` as an argument to the function. then you won't need to remove `x` and add it back. Another option is to remove duplicates at the end...

Reply:
Yeah I agree there is an inefficiency.

Actually, I think the best solution for efficiency is to split this case into a new proof rule (i.e. QUANT_ELIM_SHADOW), and make quant_unused_vars assume no shadowing.